### PR TITLE
Validate bplan image downloaded in API serializer

### LINKED ISSUE #346: Bplan: Image size restrictions are not checked when imported with the api
The project form limits image to a minimum size. Those restrictions are not checked when importing images with the api.
If a smaller image is already imported, the project can not be edited in the dashboard.

### PR DESCRIPTION
**Pull request by 2e2a**
Fixes #346

### CODE REVIEW

File: apps/bplan/serializers.py [hunk 1]
```diff
@@ -104,12 +102,35 @@ def _get_absolute_url(self, bplan):
...
+            file_name = os.path.join(settings.MEDIA_ROOT, file_path)
+            file_dir = os.path.dirname(file_name)
+            os.makedirs(file_dir, exist_ok=True)
+
+            request.urlretrieve(url, file_name)
+        except:
+            self._remove_image_if_exists(file_name)
+            raise serializers.ValidationError(
+                'Failed to download image {}'.format(url))
+        self._validate_image(file_name)
```
Comment (kleingeist) [inline apps/bplan/serializers.py]: why are you validating the image outside of the try block?
i suggest to remove the try/except block from the _validate_image function and just move the call in the try block

File: apps/bplan/serializers.py [hunk 2]
```diff
@@ -104,12 +102,35 @@ def _get_absolute_url(self, bplan):
...
+
+    def _validate_image(self, file_name):
+        image_file = open(file_name, "rb")
+        image = ImageFile(image_file, file_name)
+        config = settings.IMAGE_ALIASES.get('*', {})
+        config.update(settings.IMAGE_ALIASES['heroimage'])
+        try:
+            validate_image(image, **config)
+        except ValidationError as e:
+            self._remove_image_if_exists(file_name)
```
Comment (kleingeist) [inline apps/bplan/serializers.py]: i don't like the idea, that the validator tries to delete the validated image. see above

**Comment by kleingeist** [comment]
the image is not updated on PATCH

**Comment by kleingeist** [comment]
also start and enddate are not updated

File: apps/bplan/serializers.py (hunk 1 above)
Comment (2e2a) [inline apps/bplan/serializers.py]: To differentiate the error cases, without checking the exceptions. I'll change it.

**Comment by 2e2a** [comment]
Good point,  will fix that.

**Comment by kleingeist** [comment]
@2e2a i just did commit the changes to use the django storage backend. could you please review the changes and merge them if they are okay

File: apps/bplan/serializers.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import os
+import datetime
+import posixpath
```
Comment (xi) [inline apps/bplan/serializers.py]: Out of interest: Why the switch from `os.path` to `posixpath`?

Comment (kleingeist) [inline apps/bplan/serializers.py]: i did copy the _generate_image_filename logic from django where posixpath was used. i am not sure why the prefered it over os.path

**Pull request merged**